### ISSUE TITLE
How should the plugin/workflow handle 'terraform delete'?

### ISSUE THREAD

**Issue by flyinprogrammer** [jmccann/drone-terraform#13]
So the obvious workflow of branches should run terraform plan, and a merge to master might run terraform apply seems obvious to me.

But at the end of the day when I want to use terraform to spin my things down, how should that be accomplished?

If we figure this out I'll get this plugin a PR for delete support ?

**Comment by as0bu** [comment]
@flyinprogrammer I was thinking about this myself. I think that by default it should destroy the environment on a successful deployment, leave it up on failure, and then also have an option to not destroy via the .drone.yml. That's my two cents.

**Comment by Drc0w** [comment]
Hey there, I think that it could be useful to tell the plugin which command(s) has(have) to run after a plan. For example, I would like to bring a platform up, then provision it and finally destroy it. Maybe, it would be possible to specify multiple commands to run.

The use of the plugin could look like this:

```
pipeline:
  terraform:
    image: jmccann/drone-terraform
    plan: false
    command: ["apply", "destroy"]
    [...]
```

Or in the case we want to use a single command:

```
pipeline:
  terraform:
    image: jmccann/drone-terraform
    plan: false
    command: apply
    [...]
```

What do you guys think?

**Comment by edwinavalos** [comment]
@Drc0w I like the first option, as well as having the option to show after the terraform remote command is done would be nice.

**Comment by msuterski** [comment]
This is something I'll be looking for, too, shortly, but in a sightly different use case.

Our process will be:
1. Provision a temporary service
2. Run integration tests against it
3. Destroy the service

#2 is pretty much a a subset of #1. If we were to make the `command` a list, it can have a single element in it.

It also could make the `plan` option obsolete.


On the other hand, with my requirements, I'd like to add a flag that would block wiping out the `.terraform` at the end (and beginning), so that I can destroy the service after my tests run. With that `command` bing a single value would work for me, too.

If we agree on one or the other, I can spend time implementing it.

**Comment by jmccann** [comment]
Implemented with #50
